### ISSUE TITLE
Amplify.Datastore.clear() doesn't clear out local store after switching the environment

### ISSUE THREAD

**Issue by skryshi** [aws-amplify/amplify-ios#1402]
### Describe the bug

I have two environments: dev and prod. After switching the environments with amplify CLI, Amplify.Datastore.clear() does not clear local store from the previous environment. Restarting XCode and rebuilding the project does not help.

The only thing that works is to remove the app from the simulator (device), so it is installed fresh by XCode.

### Steps To Reproduce

```swift
Steps to reproduce the behavior:

1. Run 'amplify env checkout dev'
2. Run iOS app with Amplify.Datastore
3. Run 'amplify env checkout prod'
4. Run iOS app
4a. Call Amplify.Datastore.clear()
4b. Call Amplify.Datastore.query()
5. Get DataStore results from 'dev' environment.
```

### Expected behavior

I expect that `clear()` will clear out 'dev' data.

### Amplify Framework Version

1.13.4

### Amplify Categories

DataStore

### Dependency manager

Swift PM

### Swift version

5.3

### CLI version

5.3.1

### Xcode version

12.5.1

### Device

iPhone11-simulator

### iOS Version

14.5

**Comment by diegocstn** [comment]
@skryshi are you calling `DataStore.clear` as the very first DataStore operation?
If so, there's a chance the underlying engine hasn't fully started and the `.clear` operation gets just ignored.

**Comment by skryshi** [comment]
@diegocstn Yes, I am calling DataStore.clear as the very first DataStore operation. If that's not right, then what is the recommended sequence of calls to get DataStore started and ready? It's not in the docs.

ATM, I have this:

```
        let models = AmplifyModels()
        let apiPlugin = AWSAPIPlugin(modelRegistration: models)
        let dataStorePlugin = AWSDataStorePlugin(modelRegistration: models)
        do {
            try Amplify.add(plugin: apiPlugin)
            try Amplify.add(plugin: dataStorePlugin)
            try Amplify.add(plugin: AWSCognitoAuthPlugin())
        } catch {
            assert(false, "Could not initialize Amplify: \(error)")
        }
        DataStore.clear { _ in }

```

**Comment by diegocstn** [comment]
@skryshi I think the `DataStore.clear` operation shouldn't strictly depend on the status of the underlying engine. 
We'll take this as a bug/enhancement, dive deep and evaluate possible solutions aligned with the other platforms.
You can perhaps start DataStore first by using `.start` and if succeeded call `.clear`, but this approach has the unwanted side effect of kicking-off the sync engine.

**Comment by diegocstn** [comment]
@skryshi we've released a fix for this in [Amplify v1.22.0](https://github.com/aws-amplify/amplify-ios/releases/tag/v1.22.0)
